workers/worker-performance.worker.js

```diff
@@ -0,0 +1,124 @@
+    assert_equals(measures[0].startTime, marks[0].startTime, "measure's startTime is the first mark's startTime");
+}, "Performance marks and measures seem to be working correctly in workers");
+
+test(function testPerformanceClearing () {
+    performance.clearMarks();
+    assert_equals(performance.getEntriesByType("mark").length, 0, "clearMarks cleared the marks");
+    performance.clearMeasures();
+    assert_equals(performance.getEntriesByType("measure").length, 0, "clearMeasures cleared the measures");
+}, "Can use clearMarks and clearMeasures in workers");
+
+test(function testPerformanceResourceTiming () {  // Resource timing
+    var start = performance.now();
+    var xhr = new XMLHttpRequest();
+     // Do a synchronous request and add a little artificial delay
+    xhr.open("GET", "/resources/testharness.js?pipe=trickle(d0.25)", false);
+    xhr.send();
+     // The browser might or might not have added a resource performance entry for the importScripts() above; we're only interested in xmlhttprequest entries
+    var entries = performance.getEntriesByType("resource").filter(entry => entry.initiatorType == "xmlhttprequest");
+    assert_equals(entries.length, 1, "getEntriesByType(\"resource\") returns one entry with initiatorType of xmlhttprequest");
+    assert_true(!!entries[0].name.match(/\/resources\/testharness.js/), "Resource entry has loaded url as its name");
+    assert_equals(entries[0].entryType, "resource", "Resource entry has correct entryType");
+    assert_equals(entries[0].initiatorType, "xmlhttprequest", "Resource entry has correct initiatorType");
+    var currentTimestamp = start;
+    var currentTimestampName = "a time before it";
+    [
+        "startTime", "fetchStart", "requestStart", "responseStart", "responseEnd"
+    ].forEach((name) => {
+        var timestamp = entries[0][name];
+        // We want to skip over values that are 0 because of TAO securty rescritions
+        // Or else this test will fail. This can happen for "requestStart", "responseStart".
+        if (timestamp != 0) {
+            assert_greater_than_equal(timestamp, currentTimestamp, "Resource entry " + name + " is after " + currentTimestampName);
+            currentTimestamp = timestamp;
+            currentTimestampName = name;
+        }
+    });
+    assert_greater_than(entries[0].responseEnd, entries[0].startTime, "The resource request should have taken at least some time");
+    assert_approx_equals(entries[0].responseEnd - entries[0].responseStart, 250, 50, "Resource timing numbers reflect reality somewhat");
+    performance.clearResourceTimings();
+    assert_equals(performance.getEntriesByType("resource").length, 0, "clearResourceTimings cleared the resource timings");
+}, "Resource timing seems to work in workers");
+
+test(function testPerformanceSetResourceTimingBufferSize () {
+    performance.setResourceTimingBufferSize(0);
+    var xhr = new XMLHttpRequest();
```

Comment:
Hmm, I'm seeing a test failure here in Chrome. If I set a breakpoint on line 95 + 89, I break on 95 first and I see that there 2 entries returned by `getEntriesByType('resource')`, which is correct because `setResourceTimingBufferSize()` [does not trim the buffer](https://www.w3.org/TR/resource-timing-2/#dom-performance-setresourcetimingbuffersize). 

It appears that we have two tests racing here? Would it make sense to merge these two and run the clear -> setResourceTimingBufferSize as part of one test?

Reply:
What happened was the previous test() block was failing before it reached clearResourceTimings, so the buffer was never cleared.  To make this independent, I just brought it out into its own test.

Reply:
Ah, doh! Thanks, that makes sense.